effect/sys/run.ts

```diff
@@ -65,6 +65,7 @@ export const { run } = (new class Runtime implements RunContext {
         if (val.exit) {
           const applied = () => val.exit!(resolved);
           cleanup.set(val, applied);
+          this.#cache.delete(k);
```

Comment:
@harrysolovay perhaps this is fixed in your refactor.
This removes the `RpcClient` atom from the cache because it has an `exit` fn.

Is the following statement correct?
_Atoms that have been cleaned up should no longer be cached_

Reply:
All of this will change considerably in the next few days. Will look forward to syncing about this with you & @jainkrati.